Forge.TreeWalker.UnitTests/test/TreeWalkerUnitTests.cs

```diff
@@ -33,27 +33,60 @@ public class TreeWalkerUnitTests
 
         private Guid sessionId;
         private IForgeDictionary forgeState = new ForgeDictionary(new Dictionary<string, object>(), Guid.Empty, Guid.Empty);
-        private ITreeWalkerCallbacks callbacks;
+        private ITreeWalkerCallbacksV2 callbacksV2;
         private CancellationToken token;
         private TreeWalkerParameters parameters;
         private TreeWalkerSession session;
         private Dictionary<string, ForgeTree> forgeTrees = new Dictionary<string, ForgeTree>();
         private ConcurrentDictionary<string, Script<object>> scriptCache = new ConcurrentDictionary<string, Script<object>>();
 
 
-        public void TestInitialize(string jsonSchema, string treeName = null)
+        public void TestInitialize(string jsonSchema, string treeName = null, string currentNodeSkipActionContext = null)
         {
             // Initialize contexts, callbacks, and actions.
             this.sessionId = Guid.NewGuid();
             this.forgeState = new ForgeDictionary(new Dictionary<string, object>(), this.sessionId, this.sessionId);
-            this.callbacks = new TreeWalkerCallbacks();
+            this.callbacksV2 = new TreeWalkerCallbacksV2() { CurrentNodeSkipActionContext = currentNodeSkipActionContext };
+            this.token = new CancellationTokenSource().Token;
+
+            ForgeTree forgeTree = JsonConvert.DeserializeObject<ForgeTree>(jsonSchema);
+
+            this.parameters = new TreeWalkerParameters(
+                this.sessionId,
+                forgeTree,
+                this.forgeState,
+                this.callbacksV2,
+                this.token)
+            {
+                UserContext = new ForgeUserContext(),
+                ForgeActionsAssembly = typeof(CollectDiagnosticsAction).Assembly,
+                InitializeSubroutineTree = this.InitializeSubroutineTree,
+                TreeName = treeName,
+                Dependencies = new List<Type>() { typeof(FooEnum) },
+                ScriptCache = this.scriptCache
+            };
+
+            this.session = new TreeWalkerSession(this.parameters);
+        }
+
+        /// <summary>
+        /// Helper function to test V1 ITreeWalkerCallbacks on the contructor accepting FileSchema.
+        /// </summary>
+        /// <param name="jsonSchema">The json string content.</param>
+        /// <param name="treeName">The tree name.</param>
+        public void TestInitializeWithFile_WithV1Callbacks(string jsonSchema, string treeName = null)
```

Comment:
>TestInitializeWithFile [](start = 20, length = 22)

This isn't initializing with filePath. We can call it TestInitializeWithJsonSchema_WithV1Callbacks

Reply:
updated

---
In reply to: [506625115](https://github.com/microsoft/Forge/pull/32#discussion_r506625115) [](ancestors = 506625115)